spec/graphql/query/executor_spec.rb

```diff
@@ -68,6 +70,44 @@
   end
 
 
+  describe 'fragment resolution' do
+    let(:schema) {
+      DummyQueryType = GraphQL::ObjectType.define do
+        name "Query"
+        field :dairy do
+          type DairyType
+          resolve -> (t, a, c) {
+            mock = Minitest::Mock.new
+            mock.expect :id, 1
+          }
+        end
+      end
+
+      GraphQL::Schema.new(query: DummyQueryType, mutation: MutationType)
+    }
+    let(:variables) { nil }
+    let(:query_string) { %|
+      query getDairy {
+        dairy {
+          id
+          ... on Dairy {
+            id
+          }
+          ...repetativeFragment
+        }
+      }
+      fragment repetativeFragment on Dairy {
+        id
+      }
+    |}
+
+    it 'resolves each field only one time, even when present in multiple fragments' do
+      result
```

Comment:
Sorry to drag this on -- could you add an assertion that the `result` _is_ what you're expecting? 

I'm worried there might be an error in there, but it's being rescued, then put into `result["errors"]`. Just an assertion that 

```ruby
expected = {"data" => {
  "dairy" => { "id" => "??" }
}}
assert_equal(expected, result)
```